src/apps/chifra/pkg/abi/load_abi.go

```diff
@@ -0,0 +1,236 @@
+package abi
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/cache"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/config"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/file"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// Where to find know ABI files
+var knownAbiSubdirectories = []string{
+	"known-000", "known-005", "known-010", "known-015",
+}
+
+type AbiInterfaceMap = map[string]*types.SimpleFunction
+
+// getMapKey turns encoding into lower case, which can be used
+// as a key in AbiInterfaceMap
+func getMapKey(encoding string) string {
+	return strings.ToLower(encoding)
+}
+
+// LoadAbiFromJsonFile loads _standard_ JSON ABI, that is the one without encodings
+// and signatures. We compute these values.
+func LoadAbiFromJsonFile(filePath string, destination AbiInterfaceMap) (err error) {
+	file, err := os.OpenFile(filePath, os.O_RDONLY, 0)
+	if err != nil {
+		return
+	}
+
+	return fromJson(file, path.Base(file.Name()), destination)
+}
+
+func fromJson(reader io.Reader, abiSource string, destination AbiInterfaceMap) (err error) {
+	// Compute encodings, signatures and parse file
+	loadedAbi, err := abi.JSON(reader)
+	if err != nil {
+		return
+	}
+
+	for _, method := range loadedAbi.Methods {
+		function := types.FunctionFromAbiMethod(&method, abiSource)
+		destination[getMapKey(function.Encoding)] = function
+	}
+
+	for _, ethEvent := range loadedAbi.Events {
+		event := types.FunctionFromAbiEvent(&ethEvent, abiSource)
+		destination[getMapKey(event.Encoding)] = event
```

Comment:
Is it possible to make sure the Encoding is already lower case when it's loaded? Not sure that makes sense, just curious.